### ISSUE TITLE
Does `--headers` flag prevent to set default extensions?

### ISSUE THREAD

**Issue by ups-ishii** [cpplint/cpplint#123]
**Summary:**
Running `cpplint` with `--headers` flag seems to prevent default `extensions` values.

Is this a spec?

**Steps To Reproduce:**

* cpplint v1.4.4 

```
$ cat <<_EOF > main.cpp 
// Copyright 2020 foo
#include <main.h>
int main(int argc, char** argv) {
    char* h = "Hello";
    printf(h);
    return 0;
}
_EOF
$ cat <<_EOF > main.h
// Copyright 2020 foo#include <some_module/feature.h>
_EOF
$ ls | xargs cpplint --extensions=cpp --headers=h 2>&1 | grep -e "^main"
main.cpp:5:  Potential format string bug. Do printf("%s", h) instead.  [runtime/printf] [4]
main.h:0:  No #ifndef header guard found, suggested CPP variable is: _TMP_SAMPLE_MAIN_H_  [build/header_guard] [5]
```

When I run `cpplint` with `--exensions` and `--headers` , result looks good.

But when I run `cpplint` with only `--headers` flag actually prints below,
```
$ ls | xargs cpplint  --headers=h 2>&1 | grep -e "^main"
main.h:0:  No #ifndef header guard found, suggested CPP variable is: _TMP_SAMPLE_MAIN_H_  [build/header_guard] [5]
```
I expect both files are reported.
like set `--extensions=<default>,<appended by --headers>`

**Comment by tkruse** [comment]
thanks for reporting this, I believe you are correct, this is not right. Sadly the code is convoluted, so it is a bit difficult to make this behave both correct and backwards compatible.

Can you check / review this patch? https://github.com/cpplint/cpplint/pull/125

**Comment by ups-ishii** [comment]
It works for me! Thank you!

**Comment by tkruse** [comment]
Merged, I hope I can release in the coming days
